### ISSUE TITLE
Elm maker does not highlight errors

### ISSUE THREAD

**Issue by nyrkovalex** [neomake/neomake#1417]
Elm maker does not produce any highlighting in any case.

_This is my first time contributing to neomake so I might get something wrong._

It seems that neomake is ignoring maker output due to missing `bufnr`:

```
13:06:49 [D      ] [1.1:1] Skipped 2 entries without bufnr.
```

I am using NeoVim `v0.2.0` and Elm `v0.18.0`

### Expected behavior

All `elm-make` compilation erorrs are highlighted in buffer after running `:Neomake`

### Steps to reproduce

1. Create any elm file containing compilation errors
1. Run `:Neomake`

#### Neomake debug information
Async support: 1
Current filetype: elm

##### Enabled makers
For the current filetype (with :Neomake):
 - elmMake
   - args: ['--report=json', '--output=/dev/null']
   - auto_enabled: 1
   - errorformat: '%*[^"]"%f"%*\D%l: %m,"%f"%*\D%l: %m,%-G%f:%l: (Each undeclared identifier is reported only once,%-G%f:%l: for each function it appears in.),%-GIn file included from %f:%l:%c:,%-GIn file included from %f:%l:%c\,,%-GIn file included from %f:%l:%c,%-GIn file included from %f:%l,%-G%*[ ]from %f:%l:%c,%-G%*[ ]from %f:%l:,%-G%*[ ]from %f:%l\,,%-G%*[ ]from %f:%l,%f:%l:%c:%m,%f(%l):%m,%f:%l:%m,"%f"\, line %l%*\D%c%*[^ ] %m,%D%*\a[%*\d]: Entering directory %*[`'']%f'',%X%*\a[%*\d]: Leaving directory %*[`'']%f'',%D%*\a: Entering directory %*[`'']%f'',%X%*\a: Leaving directory %*[`'']%f'',%DMaking %*\a in %f,%f|%l| %m'
   - exe: 'elm-make'
   - process_output: function('neomake#makers#ft#elm#ElmMakeProcessOutput')
NOTE: you can define g:neomake_elm_enabled_makers to configure it (or b:neomake_elm_enabled_makers).

For the project (with :Neomake!): None.
NOTE: you can define g:neomake_enabled_makers to configure it.

Default maker settings:
 - remove_invalid_entries: 0
 - buffer_output: 1

##### Settings
```
g:neomake_javascript_enabled_makers = ['eslint']
g:neomake_place_signs = 1
g:neomake_typescript_checkers = ['tslint']


Windows: 0
[shell, shellcmdflag, shellslash]: ['/bin/bash', '-c', 0]
```

#### :version
```

NVIM v0.2.0
Build type: RelWithDebInfo
Compilation: /usr/bin/cc -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -Wconversion -Wp,-U_FORTIFY_SOURCE -Wp,-D_FORTIFY_SOURCE=1 -O2 -g -DDISABLE_LOG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -I/builddir/build/BUILD/neovim-0.2.0/build/config -I/builddir/build/BUILD/neovim-0.2.0/src -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/builddir/build/BUILD/neovim-0.2.0/build/src/nvim/auto -I/builddir/build/BUILD/neovim-0.2.0/build/include
Compiled by mockbuild

Optional features included (+) or not (-): +acl   +iconv    +jemalloc +tui      
For differences from Vim, see :help vim-differences

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

```

#### :messages
```
"index.elm" 40L, 605C
```

### Neomake logfile
```
13:06:49 [D      ] [1.-:1] Running makers: elmMake (auto).
13:06:49 [V      ] [1.1:1] Starting async job: ['elm-make', '--report=json', '--output=/dev/null', '/home/nyrkovalex/Projects/elm-hello/index.elm'].
13:06:49 [D      ] [1.1:1] cwd: /home/nyrkovalex/Projects/elm-hello.
13:06:49 [D +0.10] [1.1:1] stdout: elmMake: ['[{"tag":"NAMING ERROR","overview":"Cannot find pattern `Decrement`","subregion":null,"suggestions":["Decrementx","Increment"],"details":"Maybe you want one of the following?\n\n    Decrementx\n    Increment","region":{"start":{"line":28,"column":5},"end":{"line":28,"column":14}},"type":"error","file":"/home/nyrkovalex/Projects/elm-hello/index.elm"},{"tag":"NAMING ERROR","overview":"Cannot find variable `Decrement`","subregion":null,"suggestions":["Decrementx","Increment"],"details":"Maybe you want one of the following?\n\n    Decrementx\n    Increment","region":{"start":{"line":37,"column":24},"end":{"line":37,"column":33}},"type":"error","file":"/home/nyrkovalex/Projects/elm-hello/index.elm"}]', ''].
13:06:49 [D      ] [1.1:1] exit: elmMake: 1.
13:06:49 [D      ] [1.1:1] elmMake: processing 1 lines of output.
13:06:49 [D      ] [1.1:1] Processing 2 entries.
13:06:49 [D      ] [1.1:1] Creating location list.
13:06:49 [D      ] File-level errors cleaned in buffer 1.
13:06:49 [D      ] [1.1:1] Skipped 2 entries without bufnr.
13:06:49 [D      ] Skipping User autocmd NeomakeCountsChanged: no hooks.
13:06:49 [D      ] [1.1:1] elmMake: completed with exit code 1.
13:06:49 [D      ] [1.1:1] Cleaning jobinfo.
13:06:49 [D      ] Skipping User autocmd NeomakeJobFinished: no hooks.
13:06:49 [D      ] Skipping User autocmd NeomakeFinished: no hooks.
13:06:49 [D      ] [1.-:-] Cleaning make info.
```

**Comment by nyrkovalex** [comment]
I've created a pull request #1418 adding `bufnr` property to elm maker error dictionary.
Don't know if there is a better way to resolve it thoguh.

**Comment by blueyed** [comment]
Fixed by #1418 I assume.
